Feature/com 226 create new intro blocks

SireeKoolenWijkstra opened the pull request:
- [x] New intro block with an image and a button is made. 
- Image is visible depended on props of HopepageCallToAction, since from the design it is unclear when it should be visible and when not. It is not visible in mobile version.
- text of intro block is dependend on prop of HomepageCallToAction since from the design it is onclear when it should say 'Contribute' or 'Let's get this ball rollin'. 
- [x] Homepagefeatures shows cards with an image/icon, a title and a description . 
- No more then 3 cards in a row on a wide screen.

SireeKoolenWijkstra committed:
Make the new intro blocks, text shown is dependent on whether a rolling ball is visible.

* Insert Grid with two Grid.Cells span=7 and span=5 so that total is still 12, but picture will be on the right side of the screen

* Insert Grid with Grid.Cell span=4 for Guidelines block so that no more then 3 cards next to each other will be shown
-- add all icons that Linda (the designer) send

* Replace Icon, Link, Header and Paragraph with Card for better alignment with design system
-- aspect ratio surrounding image/icons has been deleted since the images are more alike icons then images and they have a fixed width and height
-- used GithubIcon for image/ icon for card about using Git since that seems most applicable image
-- introduced check for type of icon so that GitHubIcon can be used as well

* Introduce ShareButton
-- color dependent on darkmode

* Show not Image of Amsterdam on mobile
-- clean up styling, relegate to .css files
-- decide to not implement background color and different font sizing since this story is getting quite bloated. New story COM-265 for background color and COM-266 for font-sizing.

thomasm0 commented:
Looking very nice, so nice to finally have some more colours 🥹

A few things I've noticed:

<img width="1792" alt="Screenshot 2024-10-03 at 16 04 59" src="https://github.com/user-attachments/assets/202c188a-b542-4ad4-b533-e15efa4b7f72">
<img width="904" alt="Screenshot 2024-10-03 at 16 05 39" src="https://github.com/user-attachments/assets/c3cad99f-6a17-4f28-b6a7-fc190ac6b43f">

1. The headings need some margin bottom, I think you can use this utility: https://designsystem.amsterdam/?path=/docs/utilities-css-margin--docs
2. Each guideline should have a title between the icon and description

<img width="904" alt="Screenshot 2024-10-03 at 16 05 39" src="https://github.com/user-attachments/assets/01be9b37-daf0-488b-8cc6-5009a3c77296">
<img width="345" alt="Screenshot 2024-10-03 at 16 08 37" src="https://github.com/user-attachments/assets/a56a4648-2fdd-4798-891c-9eed995550bf">

The heading button darkmode:
3. It should not have a border or its the same colour as the background
4. On hover, the font is hard to read as it's too dark; I assume it should be white/light?

<img width="899" alt="Screenshot 2024-10-03 at 16 06 15" src="https://github.com/user-attachments/assets/d324be5d-680e-475d-b1c2-a85bd4af364c">

5. There should be more margin/padding between the content and footer.


The header image (the cranes)

6. Can we get this image in higher resolution? It's kind of small so not so pretty. It would be also be great to handle this with srcset + responsively: 
https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture
https://www.smashingmagazine.com/2014/05/responsive-images-done-right-guide-picture-srcset/

thomasm0 commented on src/components/HomepageCallToAction/index.tsx:
```diff
@@ -1,22 +1,45 @@
-  </Grid.Cell>
-);
+interface HomepageCallToActionProps {
+  isBallRolling?: boolean;
```
These boolean props are unnecessary unless they handle some styling too, which I don't see

thomasm0 commented on src/components/HomepageCallToAction/index.tsx:
```diff
@@ -1,22 +1,45 @@
+        <Grid className="ams-grid_inner">
+          <Grid.Cell span={7}>
+            <Heading size="level-2">
+              {isBallRolling ? "Let's get the ball rollin'" : 'Contribute'}
```
This is confusing; I would just set a prop `title` or `heading` but in this case it's unncessary so I would remove.

SireeKoolenWijkstra replied:
Is removed, now only Contribute is shown

thomasm0 commented on src/components/HomepageFeatures/index.tsx:
```diff
@@ -28,6 +30,7 @@ const featureList: FeatureItem[] = [
   //   description: 'Code reviews are required before merging to main',
   // },
   {
+    image: 'img/Git.svg',
```
Why do we have 2 different GitHub icons? See line 4. It also looks useful to add something to invert darker logos in dark mode.

<img width="279" alt="Screenshot 2024-10-03 at 16 19 33" src="https://github.com/user-attachments/assets/ff4d0ab1-0370-4b49-90e4-91fd96112cbf">

SireeKoolenWijkstra replied:
I've inverted the githubicon and adjusted the others their colors

thomasm0 commented on src/components/HomepageFeatures/index.tsx:
```diff
@@ -28,6 +30,7 @@ const featureList: FeatureItem[] = [
   //   description: 'Code reviews are required before merging to main',
```
What's with the commented code?

SireeKoolenWijkstra replied:
Can't remember :dizzy_face:

thomasm0 commented on src/components/HomepageFeatures/index.tsx:
```diff
@@ -43,29 +46,34 @@ const featureList: FeatureItem[] = [
   //   description: 'A guideline on how to choose',
```
Commented code?

thomasm0 commented on src/components/HomepageFeatures/index.tsx:
```diff
@@ -74,25 +82,27 @@ const featureList: FeatureItem[] = [
-    </div>
+    <Grid.Cell key={uuidv4()} span={4}>
+      <Card>
+        {typeof listItem.image === 'string' ? (
```
Isn't it possible to import the other SVG icons in the same way as the GitHub icon? Then you don't need this conditional and it will import SVGs natively, than as image elements.

SireeKoolenWijkstra replied:
Done this

thomasm0 commented on src/css/custom.css:
```diff
@@ -37,6 +37,10 @@ body {
   max-width: 1440px;
 }
 
+.ams-grid_inner {
```
Is there a reason this is not in the `ams-overrides.css` file?

SireeKoolenWijkstra replied:
Nope, was a little unclear to me that it should be in the override file. I thought it is a new class and therefor could be in the custom file. Moved it though.

thomasm0 commented on src/pages/index.module.css:
```diff
@@ -20,6 +20,12 @@
+  color: #000000;
+}
+
+
```
extra blank line 😱🙀

SireeKoolenWijkstra replied:
:face_in_clouds:

thomasm0 commented on src/theme/Button/index.tsx:
Why is the file called `Button/index.tsx` but the component it exports is called `ShareButton`?

thomasm0 replied:
I don't think this component is necessary. I would just a Link component in `src/components/HomepageCallToAction/index.tsx`, and steal the class names from [this element](https://designsystem.amsterdam/?path=/story/components-navigation-link--on-dark-background) (or the button) and made add some custom styles to remove the underline/style as our design.

SireeKoolenWijkstra replied:
Used a Link component.

thomasm0 commented on static/img/aanzicht_Amsterdam.png:
As mentioned in my [other comment](https://github.com/Amsterdam/ee-docs/pull/44#issuecomment-2391541840), can we get this in a higher resolution?

SireeKoolenWijkstra replied:
Did it.

SireeKoolenWijkstra committed:
WIP resolve feedback

SireeKoolenWijkstra committed:
Fix link and distance to footer

- update design system so that cause for error can be excluded

SireeKoolenWijkstra committed:
solve error in Footer

SireeKoolenWijkstra committed:
WIP can not get the image right

thomasm0 committed:
Feature/com 226 thomas fixes (#65)

* optimized images

* move global styling for homepage to css modules and fix for image path

SireeKoolenWijkstra committed:
WIP processing feedback

SireeKoolenWijkstra committed:
invert githubicon in dark mode and change colors of other icons

SireeKoolenWijkstra committed:
Fix rebase of develop into branch

- include fix of filterin colors in dark mode

SireeKoolenWijkstra committed:
typo in filename corrected

thomasm0 commented on src/components/HomepageFeatures/index.tsx:
```diff
@@ -1,99 +1,107 @@
+              })}
+            <Heading size="level-4">
+              {listItem.title}
+              <Card.Link href={listItem.to} />
```
In design-system, the Card example wraps the Heading text in a `Card.Link`, is there a reason for not doing that here?

design-system:
```tsx
<Card>
  <Heading size="level-4">
    <Card.Link href="/">
      Kennisgevingen en bekendmakingen
    </Card.Link>
  </Heading>
  <Paragraph>
    Kennisgevingen en bekendmakingen van de gemeente Amsterdam van de laatste 8 weken.
  </Paragraph>
</Card>
```

Otherwise I would just wrap the <Heading> and <Paragraph> in an anchor, that seems to work. Also link titles should be color `#004699` and on hover, underlined. In dark mode the color is white.

SireeKoolenWijkstra replied:
The color was added automatically when I used `Link` from the design system so I guess that's the correct color. 

The `paragraph` becomes also a link when you wrap the `heading` in a `card.link`, but doesn't get the hover status (underlined). Kept it like that since the design system thinks that's the way forward.

thomasm0 reviewed (changes requested):
Woohoo, we are almost there. Just two things left:
- The 'Guidelines' heading between the 'Contribute' intro bock and links is missing.

![Screenshot 2024-12-12 at 15 27 54](https://github.com/user-attachments/assets/5aea3ab9-da0d-4850-a785-79a907ba031c)

- The <Card.Link> styling (see my other comment)

SireeKoolenWijkstra committed:
WIP add heading Guidelines in another grid as that will fill out the whole line

SireeKoolenWijkstra committed:
WIP still problem with positioning after Guidelines

SireeKoolenWijkstra committed:
developers.amsterdam is responsive, changes colors depeding on dark mode or not and has correct heading

thomasm0 commented on src/pages/index.tsx:
```diff
@@ -33,7 +33,7 @@ const Home: FunctionComponent = () => (
     <main className={styles.container}>
       <Screen maxWidth="wide">
-        <Grid gapVertical="small" paddingVertical="medium">
+        <Grid paddingTop="medium" paddingBottom="large" className="ams-grid_outer">
```
I think the class `ams-grid_outer` can be removed, I don't see any class/styles related with this name when I inspect in devtools.

thomasm0 reviewed (changes requested):
Alrighty we are very very very close now. I see 2 things:
1. In dark mode, the link hover color is too dark (in dark mode - keep it white on hover)
2. On tablet/medium the columns in `src/components/HomepageFeatures/index.tsx` are not even width; first one takes up more width. This can be resolved by:
   1.  In `src/components/HomepageCallToAction/index.tsx` change line 7 from `<Grid.Cell span={12}>` to `<Grid.Cell span="all">`
   2. In `src/components/HomepageFeatures/index.tsx` change `<Grid.Cell key={uuidv4()} span={{ narrow: 12, medium: 6, wide: 4 }}>` to `<Grid.Cell key={uuidv4()} span={{ narrow: 12, medium: 4, wide: 4 }}>` (Note `medium` changes from `6` to `4`. I thought these should equal 12, maybe there's something in the design-system grid docs that explains this. 🤷

SireeKoolenWijkstra committed:
fix spacing in medium screen

apparantly span doesn't have to be  12.

thomasm0 committed:
Feature/com 226 intro mobile full width (#92)

This moves the grid system container into each child component instead of having 1 global grid container. Before the intro container filled not the complete width on mobile. Also English clean up

thomasm0 reviewed (approved):
woohoo!

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExeGNlMXdnYWxvNTN2cTc3bWw2bzVzZTZvZW5mejhneTkzNnUxdndibSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/3o6ZsXHLRnkgPtEYVi/giphy.gif)
